### PR TITLE
use temp kubeconfig for fake factory

### PR DESCRIPTION
**Pull request by juanvallejo**
**Release note**:
```release-note
NONE
```

Fixes https://github.com/kubernetes/kubernetes/issues/60907

cc @deads2k @ixdy

### CODE REVIEW

File: pkg/kubectl/cmd/testing/fake.go
Comment (BenTheElder) [inline pkg/kubectl/cmd/testing/fake.go]: where does this get cleaned up? 
shouldn't it explicitly be in `tempDir` instead of relying on them to match?

Comment (ixdy) [reply pkg/kubectl/cmd/testing/fake.go]: https://golang.org/pkg/io/ioutil/#TempFile
> If dir is the empty string, TempFile uses the default directory for temporary files (see os.TempDir).

the `tempDir` and `filepath.Join` bit below may not be needed at all?

**Comment by ixdy** [comment]
Confirmed that this fixes the test on my workstation. Thanks!

**Comment by BenTheElder** [comment]
this has new non-stdlib imports so it needs a `hack/update-bazel.sh`.

File: pkg/kubectl/cmd/testing/fake.go
Comment (BenTheElder) [reply pkg/kubectl/cmd/testing/fake.go]: yes, the tempDir + join smells fishy, but if we do need them, then we should actually specify the dir here, otherwise we have some weird implementation-detail coupling below.

Comment (juanvallejo) [reply pkg/kubectl/cmd/testing/fake.go]: Thanks, cleaned this up a bit.
As for the file cleanup, I added a `Cleanup` method to the fake factory that removes the temp file.
Now I just need to call it from all callers of NewTestFactory()

**Comment by juanvallejo** [comment]
Thanks, updated bazel

File: pkg/kubectl/cmd/testing/fake.go
Comment (BenTheElder) [reply pkg/kubectl/cmd/testing/fake.go]: Great!

**Comment by juanvallejo** [comment]
cc @soltysh

File: pkg/kubectl/cmd/testing/fake.go
Comment (deads2k) [inline pkg/kubectl/cmd/testing/fake.go]: don't expose wholely private methods.  Someone will mess up

File: pkg/kubectl/cmd/testing/fake.go
Comment (deads2k) [inline pkg/kubectl/cmd/testing/fake.go]: don't expose wholely private methods. Someone will mess up

File: pkg/kubectl/cmd/testing/fake.go
Comment (deads2k) [inline pkg/kubectl/cmd/testing/fake.go]: no globals!

**Comment by deads2k** [comment]
Need to remove globals before merge.

**Comment by juanvallejo** [comment]
@deads2k thanks, review comments addressed

File: pkg/kubectl/cmd/testing/fake.go
Comment (deads2k) [inline pkg/kubectl/cmd/testing/fake.go]: this doesn't seem to make sense.  Why don't you just return a loadingrules and the tempfile you created, then leave it to the `New` method to decide how to wire it.

File: pkg/kubectl/cmd/label_test.go
Comment (soltysh) [inline pkg/kubectl/cmd/label_test.go]: Use `k` to print the test name, rather than args.

File: pkg/kubectl/cmd/set/set_image_test.go
Comment (soltysh) [inline pkg/kubectl/cmd/set/set_image_test.go]: Give names, please, apiPrefix is bad here.

File: pkg/kubectl/cmd/apply_test.go
Comment (soltysh) [inline pkg/kubectl/cmd/apply_test.go]: Name please.

File: pkg/kubectl/cmd/apply_test.go
Comment (soltysh) [inline pkg/kubectl/cmd/apply_test.go]: Name please.

File: pkg/kubectl/cmd/apply_test.go
Comment (soltysh) [inline pkg/kubectl/cmd/apply_test.go]: Name please.

File: pkg/kubectl/cmd/apply_test.go
Comment (soltysh) [inline pkg/kubectl/cmd/apply_test.go]: Name please.

File: pkg/kubectl/cmd/apply_test.go
Comment (soltysh) [inline pkg/kubectl/cmd/apply_test.go]: Name.

File: pkg/kubectl/cmd/apply_test.go
Comment (soltysh) [inline pkg/kubectl/cmd/apply_test.go]: Name

File: pkg/kubectl/cmd/apply_test.go
Comment (soltysh) [inline pkg/kubectl/cmd/apply_test.go]: Name

File: pkg/kubectl/cmd/apply_test.go
Comment (soltysh) [inline pkg/kubectl/cmd/apply_test.go]: Name, or index at least

**Review by soltysh — CHANGES_REQUESTED** [review]
Fill in the names for tests, or at least indexes of these tests, so that it's clear what's failing.

**Comment by juanvallejo** [comment]
@soltysh thanks, will address comments.

This pull is not ready to merge just yet.
There are still some pieces trying to read from a user's config. I believe this is what may be causing bazel tests to fail. Looking into this.

**Comment by juanvallejo** [comment]
@soltysh thanks for the review, comments addressed

**Comment by BenTheElder** [comment]
> I believe this is what may be causing bazel tests to fail. Looking into this.

It probably is -- bazel or not CI doesn't have a user config available to load, tests are all containerized.

**Comment by juanvallejo** [comment]
@BenTheElder thanks, turned out to be loading rules. This should now be fixed

**Review by soltysh — APPROVED** [review]
/lgtm

**Pull request merged**
